memory.go

```diff
@@ -130,7 +133,10 @@ func (p *MemProvider) Read(sid string) (_ RawStore, err error) {
 	// Create a new session.
 	p.lock.Lock()
 	defer p.lock.Unlock()
-
+	if ok {
+		p.list.Remove(e)
+		delete(p.data, sid)

```

Comment:
This seems not necessary? `p.data[sid] = p.list.PushBack(s)` would override the value anyway.